generated/Normalizer/BlueprintNormalizer.php

```diff
@@ -0,0 +1,560 @@
+                        $value_2 = $value_1;
+                    } elseif (is_array($value_1) and isset($value_1['resource']) and isset($value_1['path'])) {
+                        $value_2 = $this->denormalizer->denormalize($value_1, 'WordPress\\Blueprints\\Generated\\Model\\FilesystemResource', 'json', $context);
+                    } elseif (is_array($value_1) and isset($value_1['resource']) and isset($value_1['contents'])) {
+                        $value_2 = $this->denormalizer->denormalize($value_1, 'WordPress\\Blueprints\\Generated\\Model\\InlineResource', 'json', $context);
+                    } elseif (is_array($value_1) and isset($value_1['resource']) and isset($value_1['slug'])) {
+                        $value_2 = $this->denormalizer->denormalize($value_1, 'WordPress\\Blueprints\\Generated\\Model\\CoreThemeResource', 'json', $context);
+                    } elseif (is_array($value_1) and isset($value_1['resource']) and isset($value_1['slug'])) {
+                        $value_2 = $this->denormalizer->denormalize($value_1, 'WordPress\\Blueprints\\Generated\\Model\\CorePluginResource', 'json', $context);
+                    } elseif (is_array($value_1) and isset($value_1['resource']) and isset($value_1['url'])) {
+                        $value_2 = $this->denormalizer->denormalize($value_1, 'WordPress\\Blueprints\\Generated\\Model\\UrlResource', 'json', $context);
+                    }
+                    $values_1[] = $value_2;
+                }
+                $object->setPlugins($values_1);
+            }
+            if (\array_key_exists('siteOptions', $data)) {
+                $object->setSiteOptions($this->denormalizer->denormalize($data['siteOptions'], 'WordPress\\Blueprints\\Generated\\Model\\BlueprintSiteOptions', 'json', $context));
+            }
+            if (\array_key_exists('phpExtensionBundles', $data)) {
+                $values_2 = [];
+                foreach ($data['phpExtensionBundles'] as $value_3) {
+                    $values_2[] = $value_3;
+                }
+                $object->setPhpExtensionBundles($values_2);
+            }
+            if (\array_key_exists('steps', $data)) {
+                $values_3 = [];
+                foreach ($data['steps'] as $value_4) {
+                    $value_5 = $value_4;
+                    if (is_array($value_4) and isset($value_4['step']) and isset($value_4['slug'])) {
+                        $value_5 = $this->denormalizer->denormalize($value_4, 'WordPress\\Blueprints\\Generated\\Model\\ActivatePluginStep', 'json', $context);
+                    } elseif (is_array($value_4) and isset($value_4['step']) and isset($value_4['slug'])) {
+                        $value_5 = $this->denormalizer->denormalize($value_4, 'WordPress\\Blueprints\\Generated\\Model\\ActivateThemeStep', 'json', $context);
+                    } elseif (is_array($value_4) and isset($value_4['step']) and isset($value_4['fromPath']) and isset($value_4['toPath'])) {
```

Comment:
This pattern matching requires all the `Step` keys to be in place, not just the `step` property. It will not produce useful error messages on incomplete input.

Also, it uses `and` 😱